OpenSubdiv fails to compile CUDA kernels with CUDA 7.0+

When building OpenSubdiv with CUDA 7.0, generating the CUDA kernels fails with this message:
```
[ 87%] Building NVCC (Device) object opensubdiv/CMakeFiles/osd_static_gpu.dir/osd/./osd_static_gpu_generated_cudaKernel.cu.o
nvcc fatal   : Value 'compute_11' is not defined for option 'gpu-architecture'
CMake Error at osd_static_gpu_generated_cudaKernel.cu.o.cmake:206 (message):
```
From version 7.0 on, the 1.x compute architectures are no longer supported, so the CMake scripts should probably detect that case and use compute_20 instead.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -475,6 +475,22 @@ if(CUDA_FOUND)
         -DOPENSUBDIV_HAS_CUDA
     )
     set(OSD_GPU TRUE)
+
+    if (UNIX)
+        list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC )
+        if (NOT DEFINED OSD_CUDA_NVCC_FLAGS)
+            if (CUDA_VERSION_MAJOR LESS 6)
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_11 )
+            else()
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_20 )
+            endif()
+        endif()
+    endif()
+
+    if (DEFINED OSD_CUDA_NVCC_FLAGS)
+        list( APPEND CUDA_NVCC_FLAGS ${OSD_CUDA_NVCC_FLAGS})
+    endif()
+
 else()
     if (NOT NO_CUDA)
         message(WARNING

--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -105,9 +105,6 @@ if (NOT NO_LIB)
 
     if( CUDA_FOUND )
         include_directories( "${CUDA_INCLUDE_DIRS}" )
-        if (UNIX)
-            list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_11 )
-        endif()
     endif()
 
 

--- a/opensubdiv/osd/CMakeLists.txt
+++ b/opensubdiv/osd/CMakeLists.txt
@@ -294,10 +294,6 @@ if( CUDA_FOUND )
             cudaD3D11VertexBuffer.h
         )
     endif()
-
-    if (UNIX)
-        list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_11 )
-    endif()
 endif()
 
 list(APPEND DOXY_HEADER_FILES ${CUDA_PUBLIC_HEADERS})

